skyshop/src/main/java/org/skypro/skyshop/controller/ShopController.java

```diff
@@ -0,0 +1,64 @@
+package org.skypro.skyshop.controller;
+
+import org.skypro.skyshop.model.product.Product;
+import org.skypro.skyshop.model.search.Searchable;
+import org.skypro.skyshop.service.SearchService;
+import org.skypro.skyshop.service.StorageService;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@RequestMapping
+public class ShopController {
+
+    private final StorageService storageService;
+    private final SearchService searchService;
+
+    public ShopController(StorageService storageService, SearchService searchService) {
+        this.storageService = storageService;
+        this.searchService = searchService;
+    }
+
+    // Возвращаем JSON список продуктов
+    @GetMapping("/products")
+    public List<? extends Searchable> getProducts() {
+        return new ArrayList<>(storageService.getProducts());
+    }
```

Comment:
нет никакого смысла оборачивать в ArrayList, лучше просто заменить тип на более общий
```suggestion
    @GetMapping("/products")
    public Collection<? extends Searchable> getProducts() {
        return storageService.getProducts();
    }
```